### PR TITLE
Feature/303 modularize spawner fixtures

### LINKED ISSUE #303: modularize spawner fixtures
I want to modularize fixtures for spawner tests in order to be able to reuse them.

### PR DESCRIPTION
**Pull request by CR1337**
Fixes #303 

Fixtures for Spawner, Schedules and ScheduleStrategies are now modularized. Tests are adapted to it.

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [x] Dev-branch has been merged into local branch to resolve conflicts
- [x] Tests and linter have passed AFTER local merge
- [x] Another dev reviewed and approved

### CODE REVIEW

File: tests/spawner/conftest.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    return [
+        "399ec209-5f72-47be-a821-31a177021311",
+        "2cf25596-24cf-49ee-ab44-323522bc2fc1",
+        "ccffa906-9324-42bb-afed-a0f923f7e934",
+        "f97dd0ba-5c3d-4a71-a1ac-01aca73aa8ce",
+        "67e5c841-9fcd-47c3-b33f-d856c3c0b3f4",
+        "5e47bb48-82ba-47ea-8295-c70c2c46cdba",
+        "9ac02dac-6f0f-4779-b730-c2b22e1d8258",
+        "91ce408a-5509-43f6-bb89-137feb0d14f2",
+    ]
```
Comment (instant-sky) [inline tests/spawner/conftest.py:190]: Are these just some random UUIDs?

File: tests/spawner/test_random_schedule_strategy.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    @pytest.mark.usefixtures(
+        "random_strategy", "random_strategy_spawn_ticks", "strategy_end_tick"
+    )
+    def test_spawn_at_right_ticks(
+        self,
+        random_strategy: RandomScheduleStrategy,
+        random_strategy_spawn_ticks: list[int],
+        strategy_end_tick: int,
+    ):
+        for tick in range(0, strategy_end_tick * 2):
```
Comment (instant-sky) [inline tests/spawner/test_random_schedule_strategy.py:23]: why * 2?

**Review by instant-sky — COMMENTED** [review]
Defining all required objects as fixtures is a very good change!

File: tests/spawner/test_random_schedule_strategy.py (hunk 2 above)
Comment (CR1337) [reply tests/spawner/test_random_schedule_strategy.py:23]: It's just to ensure that ticks are generated where no trains are allowed to be spawned. Then you can test if there are really no trains spawned.

File: tests/spawner/conftest.py (hunk 1 above)
Comment (CR1337) [reply tests/spawner/conftest.py:190]: yap :D

**Review by instant-sky — APPROVED** [review]

**Pull request merged**